common/src/main/java/org/apache/cassandra/sidecar/common/JmxClient.java

```diff
@@ -259,12 +259,16 @@ private Map<String, Object> buildJmxEnv()
     }
 
     @Override
-    public synchronized void close() throws IOException
+    public void close() throws IOException
     {
-        JMXConnector connector = jmxConnector;
-        if (connector != null)
+        JMXConnector connector;
+        synchronized (this)
         {
+            connector = jmxConnector;
             jmxConnector = null;
```

Comment:
Reset `connected` to false too?

Reply:
Good catch. This would need to be reset to false to prevent false-positives from `checkConnection`. Will add.